onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -1830,6 +1831,24 @@ static bool ConvTransposeNeedFallbackToCPU(const onnxruntime::Node& node) {
         }
       }
     }
+
+    if ("auto_pad" == attr_name &&
+        ::ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRING == attr_value.type()) {
+      auto& auto_pad_attr = attr_value.s();
+      ORT_ENFORCE(auto_pad_attr == "SAME_UPPER" || auto_pad_attr == "SAME_LOWER" ||
+                      auto_pad_attr == "VALID" || auto_pad_attr == "NOTSET",
+                  "auto_pad must be either NOTSET, VALID, SAME_UPPER, SAME_LOWER");
+
+      // If auto_pad is SAME_UPPER or SAME_LOWER, pads will be computed dynamically at runtime
+      // based on the provided input shape. This may or may not lead to symmetric padding.
+      // If it turns out to be asymmetric padding, CuDNN will return a cryptic unfriendly error message.
+      // So drop down the node to CPU if auto_pad is SAME_UPPER or SAME_LOWER even if it may lead to
+      // symmetric padding.
+      // TODO: Remove this after we have supported asymmetric padding in the CUDA ConvTranspose kernel
+      if (auto_pad_attr == "SAME_UPPER" || auto_pad_attr == "SAME_LOWER") {
```

Comment:
Can we log something saying why the fallback is taking place to CPU (unless it's not logged anywhere else)?

Reply:
Good catch - I actually meant to do that at the time of writing the "if", but it seems like I have forgotten. Let me add the logs, thanks.

Reply:
It should be logged somewhere else that the node is falling back - but it may be meaningful to add a more informative log saying why this node is falling back.

Reply:
Yeah, the layer that logs above this will lose the reason making users wonder.

Reply:
Added logs